src/Http/Client/Adapter/Stream.php

```diff
@@ -268,7 +269,7 @@ protected function _send(RequestInterface $request): array
         fclose($this->_stream);
 
         if ($timedOut) {
-            throw new HttpException('Connection timed out ' . $url, 504);
+            throw new NetworkException('Connection timed out ' . $url, $request);
```

Comment:
This change will break the behavior where an http request to a backend timing out will result in the app also responding with a 504.

Reply:
I can make `NetworkException` extend `HttpException` and set `504` status code but personally I don't think a timeout on an internal request should bubble up as `504` for the parent request. It should still be a 500 IMO.

Reply:
I checked a couple of guzzle wrappers for PSR18 support and couple of standalone PSR18 implementations and non set a `504` code.

Reply:
Ok. Then perhaps our code should change too.